starter-code/js/index.js

```diff
@@ -1,11 +1,64 @@
-var $cart = document.querySelector('#cart tbody');
-var $calc = document.getElementById('calc');
+const $cart = document.querySelector("#cart tbody");
+const $calc = document.getElementById("calc");
 
-function updateSubtotal($product) {
-  // Iteration 1.1
+function updateSubtotal(product) {
+  const unitPrice = product.querySelector(".pu span").innerText;
+  const quantity = product.querySelector(".qty input").value;
+  let subTotal = unitPrice * quantity;
+  product.querySelector(".subtot span").innerText = `${subTotal}`;
+  return subTotal;
 }
 
 function calculateAll() {
-  // Iteration 1.2
+  let count = 0;
+  document.querySelectorAll(".product").forEach(element => {
+    count += updateSubtotal(element);
+  });
+  console.log(count);
+  document.querySelector("h2 span").innerText = `${count}`;
 }
-$calc.onclick = calculateAll;
\ No newline at end of file
+
+document.querySelectorAll(".btn-delete").forEach(function(button) {
+  button.onclick = deleteButton;
+});
+
+function deleteButton(e) {
+  const product = e.currentTarget.parentNode.parentNode;
+  $cart.removeChild(product);
+}
+const createButton = document.querySelector("#create");
+createButton.onclick = createProduct;
+
+function createProduct(e) {
+  const npName = document.querySelector("#nName").value;
+  const npPrice = parseFloat(document.querySelector("#nPrice").value).toFixed(
+    2
+  );
+  const newProduct = document.createElement("tr");
+  newProduct.className = "product";
+  newProduct.innerHTML = `
```

Comment:
You can get around having to do this by using: cloneNode, however this is totally fine as well:)

Reply:
Thank you Markus! I decided not to use cloneNode intentionally. in case we deleted all the items and there's nothing left to clone :)